pydy/viz/server.py

```diff
@@ -1,29 +1,77 @@
 #!/usr/bin/env python
 
+from __future__ import print_function
+
 import os
+import signal
+import socket
 import webbrowser
 import BaseHTTPServer
 from SimpleHTTPServer import SimpleHTTPRequestHandler
 
 
-def run_server(port=8000,scene_file="Null"):
-    #change dir to static first.
-    os.chdir("static/")
-    HandlerClass = SimpleHTTPRequestHandler
-    ServerClass  = BaseHTTPServer.HTTPServer
-    Protocol     = "HTTP/1.0"
-    server_address = ('127.0.0.1', port)
-    HandlerClass.protocol_version = Protocol
-    httpd = ServerClass(server_address, HandlerClass)
-    sa = httpd.socket.getsockname()
-    print "Serving HTTP on", sa[0], "port", sa[1], "..."
-    print "hit ctrl+c to stop the server.."
-    print "To view visualization, open:\n"
-    url = "http://localhost:"+str(sa[1]) + "/index.html?load="+scene_file
-    print url
-    webbrowser.open(url)
-    httpd.serve_forever()
-
-
-if __name__ == "__main__":
-    run_server()
+class StoppableHTTPServer(BaseHTTPServer.HTTPServer):
+
+    def server_bind(self):
+        BaseHTTPServer.HTTPServer.server_bind(self)
+        self.socket.settimeout(1)
+        self.run = True
+
+    def get_request(self):
+        while self.run:
+            try:
+                sock, addr = self.socket.accept()
+                sock.settimeout(None)
+                return (sock, addr)
+            except socket.timeout:
+                pass
+
+    def stop(self):
+        self.run = False
+
+    def serve(self):
+        while self.run:
+            self.handle_request()
+
+
+class Server(object):
+    """
+    Parameters
+    ----------
+    port : integer
+        Defines the port on which the server will run.
+    scene_file : name of the scene_file generated for visualization
+        Used here to display the url
+    directory : path of the directory which contains static and scene files.
+        Server is started in this directory itself.
+
+    """
+    def __init__(self, port=8000, scene_file="Null", directory="static/"):
```

Comment:
It runs from static directory present in current directory but Yeah, i too think that this should be a required arg.
Also, scene_file is used to calculate the url and print it, as a result it can't be None, It must be Null (a string)